### ISSUE TITLE
Is there any important reason why the relay nodes are not synchronized?

### ISSUE THREAD

**Issue by supersensen** [nostr-protocol/nips#338]

**Comment by fiatjaf** [comment]
Yes. That would require everybody to store all events, which is impossible.

**Comment by supersensen** [comment]
Should we at least unify user data? 
For example, the two relays are not connected. One user registers with the same private key on the two relays. One user's age information is 18 and the other user's age information is 19. What does the client think of the user's age? @fiatjaf

**Comment by fiatjaf** [comment]
It can use the event with the latest `created_at` -- or it can decide in any way it wants.

**Comment by supersensen** [comment]
Ok, I'll try to understand it. Thank you!

**Comment by mikedilger** [comment]
Think of relays as fairly dumb storage that helps you distribute your events.  Think of your events as your official statements, digitally signed, unforgeable, totally under your control and not that of the relays.  If you say you are 18 in one statement and push that to one relay, and 19 in a different statement to a different relay, it doesn't matter which relays those statements are on, the latest statement supersedes the previous one.  It just might be that some clients only find the earlier one and have outdated information.  Such is life.

Registering on a relay is just an arrangement to allow you to store your events there, the relay doesn't care about what the content of those events is. Most relays don't look to see that you are 18 or 19, they are just dumb storage.

I hope that gives some perspective.

**Comment by weex** [comment]
I had the same question and thanks @supersensen for raising it. If it's answered to your satisfaction, you may close the issue to help keep the tracker tidy.

**Comment by IngwiePhoenix** [comment]
I think that, to a certain degree, there should be a network of relays communicating with one another to create redundancy/resilience.

I have only been running my `strfry` instance for about two days and it almost accumulated 1GB, only by dumping events from nos.lol and nos.wine into it's DB too (was experimenting with using a single WSS connection on my phone - internet around here is pretty crap...).

However, this is something I have actually been wondering about too and whilst I don't think it merits a formal NIP spec, it *should* be explained in adjanced documentation.

More specifically, this:

- Pros and Cons of one user connecting to few or many relays.
- When relays should and should not sync data.
- How users can isolate in certain communities by the choice of their relay/s (i.e. `myrelay.com/groupa` is only for a specific group, whilst `myrelay.com/groupb` might be for another).

As far as I understood it - from both poking around in the community and reading each NIP - there's two potential "paradigms": One, in which many relays build a network of redundancy and resilience as for when one relay fails, it hardly makes a difference; and two in which some users may only spend time on a few single relays to be part of just some communities. However, the latter falls apart if a user writes but a single event referencing another to a different relay...

**Comment by mikedilger** [comment]
First, IF there is a separate protocol between relays to share events, that would not be part of the nostr protocol. It would be a separate related protocol.  Of course relays can behave like clients and pull events from other relays over the nostr protocol.

Second, I don't think relays should mirror events from other relays.  Could you imagine if every website in the world mirrored all the webpages from all the other websites?  That cannot scale.  I think of relays similarly.  Redundancy comes about because clients use multiple relays.

If you are concerned that your client cannot find relevant events because they are not on the relays you are connected to... then yes that is a real concern.  But the solution isn't to copy events everywhere. The solution I push is for clients to figure out which relays have the relevant events that you are looking for. That was called the "gossip model" (because the gossip client tried to do that from day 1), and it was made easier with [NIP-65](https://github.com/nostr-protocol/nips/blob/master/65.md ) which is now implemented in at least gossip client, coracle, and snort.
